src/models/PropStore.ts

```diff
@@ -0,0 +1,56 @@
+import { InjectionKey, reactive } from 'vue'
+import { isWithComponent, isWithComponents } from '@/types/createRouteOptions'
+import { ResolvedRoute } from '@/types/resolved'
+import { Route } from '@/types/route'
+import { MaybePromise } from '@/types/utilities'
+
+export const propStoreKey: InjectionKey<PropStore> = Symbol()
+
+type ComponentProps = { id: string, name: string, props?: (params: Record<string, unknown>) => MaybePromise<Record<string, unknown>> }
```

Comment:
I think this is true
```suggestion
type ComponentProps = { id: string, name: string, props?: (params: Record<string, unknown>) => unknown }
```